test/framework/src/main/java/org/opensearch/test/InternalTestCluster.java

```diff
@@ -2168,7 +2169,13 @@ public String getClusterManagerName() {
     public String getClusterManagerName(@Nullable String viaNode) {
         try {
             Client client = viaNode != null ? client(viaNode) : client();
-            return client.admin().cluster().prepareState().get().getState().nodes().getClusterManagerNode().getName();
+            DiscoveryNode masterNode = client.admin().cluster().prepareState().get().getState().nodes().getClusterManagerNode();
+            if (masterNode == null) {
+                throw new ClusterManagerNotDiscoveredException("Cluster manager node not discovered");
+            }
+            return masterNode.getName();
```

Comment:
Just a naming nitpick here...

```suggestion
            DiscoveryNode clusterManagerNode = client.admin().cluster().prepareState().get().getState().nodes().getClusterManagerNode();
            if (clusterManagerNode == null) {
                throw new ClusterManagerNotDiscoveredException("Cluster manager node not discovered");
            }
            return clusterManagerNode.getName();
```